### ISSUE TITLE
SIGINT not being sent on startOrReload

### ISSUE THREAD

**Issue by CareLuLu-Gabriel** [Unitech/pm2#4300]
## What's going wrong?

PM2 on cluster sends SIGINT on `pm2 stop pm2.json` but not on `pm2 startOrReload pm2.json`. I tested both Mac OS and Amazon Linux with Node v12.3.1.

## How could we reproduce this issue?

pm.json
```
{
	"apps": [{
		"script": "index.js",
		"instances": "2",
		"exec_mode": "cluster",
		"wait_ready": true,
		"listen_timeout": 300000,
		"kill_timeout": 30000,
		"autorestart": true,
		"max_memory_restart": "1G",
		"cron_restart": "1 0 * * *"
	}],
}
```

1) Run `pm2 startOrReload pm2.json`.
2) Run `pm2 startOrReload pm2.json` and it should get stuck.

## Supporting information

```
--- PM2 report ----------------------------------------------------------------
Date                 : Thu May 30 2019 21:09:49 GMT-0700 (Pacific Daylight Time)
===============================================================================
--- Daemon -------------------------------------------------
pm2d version         : 3.5.1
node version         : 12.3.1
node path            : /Users/me/server/node_modules/.bin/pm2
argv                 : /Users/me/.nvm/versions/node/v12.3.1/bin/node,/Users/me/server/node_modules/pm2/lib/Daemon.js
argv0                : /Users/me/.nvm/versions/node/v12.3.1/bin/node
user                 : gabrielmarques
uid                  : 501
gid                  : 20
uptime               : 0min
===============================================================================
--- CLI ----------------------------------------------------
local pm2            : 3.5.1
node version         : 12.3.1
node path            : /Users/me/server/node_modules/.bin/pm2
argv                 : /Users/me/.nvm/versions/node/v12.3.1/bin/node,/Users/me/server/node_modules/.bin/pm2,report
argv0                : /Users/me/.nvm/versions/node/v12.3.1/bin/node
user                 : gabrielmarques
uid                  : 501
gid                  : 20
===============================================================================
--- System info --------------------------------------------
arch                 : x64
platform             : darwin
type                 : Darwin
cpus                 : Intel(R) Core(TM) i5-4308U CPU @ 2.80GHz
cpus nb              : 4
freemem              : 129126400
totalmem             : 8589934592
home                 : /Users/gabrielmarques
===============================================================================
--- PM2 list -----------------------------------------------
┌──────────┬────┬─────────┬─────────┬───────┬────────┬─────────┬────────┬─────┬────────────┬────────────────┬──────────┐
│ App name │ id │ version │ mode    │ pid   │ status │ restart │ uptime │ cpu │ mem        │ user           │ watching │
├──────────┼────┼─────────┼─────────┼───────┼────────┼─────────┼────────┼─────┼────────────┼────────────────┼──────────┤
│ index    │ 0  │ 3.0.1   │ cluster │ 22161 │ online │ 1       │ 17s    │ 0%  │ 235.6 MB   │ gabrielmarques │ disabled │
│ index    │ 1  │ 3.0.1   │ cluster │ 22142 │ online │ 0       │ 27s    │ 0%  │ 120.5 MB   │ gabrielmarques │ disabled │
└──────────┴────┴─────────┴─────────┴───────┴────────┴─────────┴────────┴─────┴────────────┴────────────────┴──────────┘
===============================================================================
--- Daemon logs --------------------------------------------
/Users/me/.pm2/pm2.log last 20 lines:
PM2        | 2019-05-30T21:09:16: PM2 log: App [index:0] starting in -cluster mode-
PM2        | 2019-05-30T21:09:22: PM2 log: App [index:0] online
PM2        | 2019-05-30T21:09:22: PM2 log: App [index:1] starting in -cluster mode-
PM2        | 2019-05-30T21:09:26: PM2 log: App [index:1] online
PM2        | 2019-05-30T21:09:32: PM2 log: App [index:0] starting in -cluster mode-
PM2        | 2019-05-30T21:09:36: PM2 log: [PM2][WORKER] Registering a cron job on: 0
PM2        | 2019-05-30T21:09:36: PM2 log: [PM2][WORKER] Registering a cron job on: 1
PM2        | 2019-05-30T21:09:37: PM2 log: App [index:0] online

```

**Comment by Goodluckhf** [comment]
I have the same problem. 
@CareLuLu-Gabriel in your case you have `300` seconds `listen_timeout` so u just cant wait for `SIGINT`. I found what really happened: `process.send('ready')` emitted before worker subscribe for this event. I have fix for this

**Comment by Unitech** [comment]
https://github.com/Unitech/pm2/pull/4376 will  land on 4.0.1

**Comment by Unitech** [comment]
Available in PM2 4.1.0

```bash
npm install pm2@latest -g
pm2 update
```
